src/AppBundle/Service/RemotePushNotificationManager.php

```diff
@@ -185,7 +189,16 @@ public function send($message, $recipients, $data = [])
             return $token->getPlatform() === 'ios';
         });
 
-        $this->fcm($message, $fcmTokens, $data);
-        $this->apns($message, $apnsTokens, $data);
+        //todo temporary send both "notification+data" and "data-only" messages on android
+        // reasons:
+        // 1. backward compatibility with the old app-versions
+        // 2. app does not always handle the case when "notification+data" is delivered in
+        // the background
+        // when #2 is done and #1 is not relevant any more we should only send "data-only" messages
+
+        $this->fcm($textMessage, $fcmTokens, $data); // send "notification+data" message
+        $this->fcm(null, $fcmTokens, $data); // send "data-only" message
```

Comment:
Does this send the notification twice?

Reply:
yes, it does, but they would be of two different types "notification+data" and "data-only". There would be the following scenarios:

- current version: the behavior would stay the same, it does not handle "data-only" messages
- new version (in the foreground): the behavior would also be the same, it would handle "data-only" messages and ignore "notification+data" messages
- new version (in the background): "notification+data" message would be shown in the notification center, "data-only" would be handled only for "order:created" event - the app would be opened.

To sum up, it would be an issue only in one case, when "order:created" event is received in the background: the app would be opened and there would be also a notification in the notification center. 

It's not ideal, but I don't have any ideal solution in mind. First of all, we still need to send "notification+data" for the current app version (it would be cool to know not only platform but also an app version for each push token, but it's quite some work to make it happen), and the second, some changes are required on the app side to be able to get rid from "notification+data" message (the app needs to show notification in the notification center, when it receives a "data-only" message)